### PR TITLE
Regular expressions

### PR DESCRIPTION
**Pull request by markknol**
I tried to create a beginner section to regular expressions. I don't want to repeat whats already on the manual or internet so I just added some snippets on how to match/replace/map results.

Can someone review this?

view here: https://github.com/HaxeFoundation/code-cookbook/blob/78e800cb4dc8a496c681b5b839c7bde6f1a75d43/assets/content/cookbook/Beginner/regular-expressions.md

### CODE REVIEW

File: assets/content/cookbook/Beginner/regular-expressions.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
+# Regular expressions
+
+In Haxe a [regular expression](https://en.wikipedia.org/wiki/Regular_expression) starts with `~/` and ends with a single `/` and is of type [`Ereg`](http://api.haxe.org/EReg.html).
```
Comment (andyli) [inline assets/content/cookbook/Beginner/regular-expressions.md]: typo: `Ereg` => `EReg`

File: assets/content/cookbook/Beginner/regular-expressions.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+Add flags with ending with adding the flag after the ending `/`.
+
+```haxe
+var regexp:EReg = ~/world/i; // case insensitive matching
+
+trace(regexp.match("HELLO WORLD"));
+// true : 'world' was found in the string
+```
+
+String based regular expressions can be writting by creating an actual `EReg` instance:
```
Comment (andyli) [inline assets/content/cookbook/Beginner/regular-expressions.md]: It is a little bit weird. I would say:

Regular expressions with a dynamic pattern can be created by using the `EReg` constructor as follows:
```haxe
var world = "world";
var regexp:EReg = new EReg("hello " + world, "i"); // case insensitive matching 

trace(regexp.match("HELLO WORLD"));
// true : 'hello world' was found in the string
```

File: assets/content/cookbook/Beginner/regular-expressions.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+```haxe
+var regexp:EReg = new EReg("world", "i"); // case insensitive matching 
+
+trace(regexp.match("HELLO WORLD"));
+// true : 'world' was found in the string
+```
+
+### Replacing text
+
+Simply replacing text can be done in several ways, even without regular expressions.
```
Comment (andyli) [inline assets/content/cookbook/Beginner/regular-expressions.md]: "Simply replacing text" => "Simple text replacement"

File: assets/content/cookbook/Beginner/regular-expressions.md (hunk 2 above)
Comment (nadako) [inline assets/content/cookbook/Beginner/regular-expressions.md]: Well, the literal syntax (`~/.../`) also creates "actual EReg instance", maybe we should say something like "by using the normal `new` operator".

File: assets/content/cookbook/Beginner/regular-expressions.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+```
+
+The equivalent using regular expressions would be:
+
+```haxe
+var message = "hello world";
+var ereg:EReg = ~/hello/;
+trace(ereg.replace(message, "happy")); 
+// "happy world"
+```
```
Comment (nadako) [inline assets/content/cookbook/Beginner/regular-expressions.md]: maybe it makes sense to mention `g` flag for global replace.

**Pull request merged**